Source/UI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift

```diff
@@ -231,12 +245,22 @@ extension DefaultBillingFormViewModel: BillingFormTextFieldDelegate {
 // MARK: - Billing form view controller Delegate
 
 extension DefaultBillingFormViewModel: BillingFormViewControllerDelegate {
-    func phoneNumberIsUpdated(number: String, tag: Int) {
-        let index = BillingFormCell.phoneNumber(nil).index
-        textValueOfCellType[index] = number
+    func textFieldDidEndEditing(tag: Int) {
         updatedRow = tag
     }
 
+    func isValidPhoneMaxLength(text: String?) -> Bool {
+        validatePhoneNumberMaxLength(text: text)
+    }
+
+    func phoneNumberIsUpdated(number: Phone, tag: Int) {
+        let index = BillingFormCell.phoneNumber(nil).index
+        textValueOfCellType[index] = number.number
+        let currentCellTypeIndex = style.cells[tag].index
+        errorFlagOfCellType[currentCellTypeIndex] = !PhoneNumberValidator().validate(value: number)
+        notifyContentChangeToDelegate()
+    }
+
```

Comment:
Why was the `textFieldDidEndEditing` introduced? It seems previously this was covered within `phoneNumberIsUpdated`?